src/modules/commander/Commander.cpp

```diff
@@ -2429,7 +2429,10 @@ Commander::run()
 			// Evaluate current prearm status
 			if (!armed.armed && !status_flags.condition_calibration_enabled) {
 				bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, status, status_flags, true, false, true, 30_s);
-				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, _safety, _arm_requirements, status, false);
+
+				PreFlightCheck::arm_requirements_t arm_req_no_auth = _arm_requirements;
+				arm_req_no_auth.arm_authorization = false;
```

Comment:
I would prefer to not request arming authorization but, if we fix the mavlink log spamming by this repetitive request, it could be ok for me.

But I don't think that good thing in general cases.
Arm authorization can depend on many things, even external to the drone. It can result in a long process for the authorizer. 
Mavlink doc says that arm authorization in required to comply https://utm.arc.nasa.gov/index.shtml. I think that in this kind of system, arm authorization should be requested only when the drone really aim to arm. Else, traffic management and takeoff authorization could be perturbed.

I don't think it is a problem if the autopilot say that arming is possible but when really trying to arm, the authorizer reject the demand. It is not the autopilot responsibility.

Reply:
Yes, sure we should reduce the spamming. And we can extend the arming authorization with a flag if we really want to arm or just checking. Then we have the option to do really expensive checks only when actually trying to arm.

Reply:
I have added a flag to preArmCheck to tell if we really want to arm and use this to check arm authorization or not